src/diem/testing/suites/test_payment_command.py

```diff
@@ -414,6 +416,79 @@ def test_payment_command_sender_address_can_only_be_written_once(
         sender_account.log_events()
 
 
+@pytest.mark.parametrize(
+    "field_name",
+    [
+        "currency",
+        "amount",
+        "timestamp",
+    ],
+)
+def test_payment_command_action_can_only_be_written_once(
+    currency: str,
+    travel_rule_threshold: int,
+    stub_wallet_app: App,
+    hrp: str,
+    stub_client: RestClient,
+    target_client: RestClient,
+    field_name: str,
+) -> None:
+    """
+    Test Plan:
+
+    1. Generate a valid account identifier from receiver account as payee.
+    2. Disable stub wallet application background tasks.
+    3. Send a payment requires off-chain PaymentCommand to the receiver account identifier.
+    4. Send initial payment command to receiver wallet offchain API endpoint.
+    5. Wait for receiver's wallet application sends payment command back with receiver ready for settlement.
+    6. Update payment command action.
+    7. Send the updated payment command to receiver with sender ready for settlement.
+    8. Expect response status code is 400
+    9. Expect response `CommandResponseObject` with failure status, `command_error` error type,
+       and `invalid_overwrite` error code.
+    """
+
+    sender_account = stub_client.create_account(
+        balances={currency: travel_rule_threshold}, kyc_data=target_client.get_kyc_sample().minimum
+    )
+    receiver_account = target_client.create_account(kyc_data=stub_client.get_kyc_sample().minimum)
+    try:
+        payee = receiver_account.generate_account_identifier()
+
+        with disable_background_tasks(stub_wallet_app):
+            payment = sender_account.send_payment(currency=currency, amount=travel_rule_threshold, payee=payee)
+            txn = stub_wallet_app.store.find(Transaction, id=payment.id)
+            # send initial payment command
+            initial_cmd = stub_wallet_app.send_initial_payment_command(txn)
+
+            # wait for receiver to update payment command
+            sender_account.wait_for_event("updated_payment_command")
+            # find updated payment command
+            updated_cmd = stub_wallet_app.store.find(PaymentCommand, reference_id=initial_cmd.reference_id())
+            offchain_cmd = updated_cmd.to_offchain_command()
+
+            # update action field value
+            if field_name == "currency":
```

Comment:
Only certain fields can't be written again right, not all fields?

Reply:
field "action" only has one value "charge", can't be changed anyway, if it is changed, the error can be invalid value instead of invalid overwrite, thus no need test for it, as we can't be sure which error will be checked first by wallet application.